velox/type/Filter.h

```diff
@@ -925,22 +968,48 @@ class BigintValuesUsingHashTable final : public Filter {
         nullAllowed_ ? "with nulls" : "no nulls");
   }
 
-  bool testingEquals(const Filter& other) const final;
+  bool testingEquals(const Filter& other) const override {
+    return IntValuesUsingHashTable<int64_t>::testingEquals(other);
+  }
 
  private:
   std::unique_ptr<Filter>
   mergeWith(int64_t min, int64_t max, const Filter* other) const;
+};
 
-  static constexpr int64_t kEmptyMarker = 0xdeadbeefbadefeedL;
-  // from Murmur hash
-  static constexpr uint64_t M = 0xc6a4a7935bd1e995L;
+/// IN-list filter for int128_t data type, implemented as a hash table.
+class HugeintValuesUsingHashTable final
+    : public IntValuesUsingHashTable<int128_t> {
```

Comment:
I don't think you need to inherit `IntValuesUsingHashTable` here.  Just keep a `F14FastSet<int128_t>` and use it as hash table.  The point of hash table in `BigintValuesUsingHashTable` is to do `testValues`, which is not very beneficial for wide values like int128.

Reply:
Updated, thanks for the suggestion!